services/auth/source/oauth2/init.go

```diff
@@ -30,6 +30,11 @@ const ProviderHeaderKey = "gitea-oauth2-provider"
 
 // Init initializes the oauth source
 func Init(ctx context.Context) error {
+	// if oauth is disabled, we don't need to initialize anything
+	if !setting.OAuth2.Enabled {
```

Comment:
This is not right because line 43 and below code should be invoked even if Oauth2 provider doesn't enabled.